src/_pytest/python_api.py

```diff
@@ -340,14 +341,14 @@ def _repr_compare(self, other_side: Sequence[float]) -> list[str]:
             zip(approx_side_as_map, other_side)
         ):
             if approx_value != other_value:
-                abs_diff = abs(approx_value.expected - other_value)
-                max_abs_diff = max(max_abs_diff, abs_diff)
-                if other_value == 0.0:
-                    max_rel_diff = math.inf
-                else:
-                    max_rel_diff = max(max_rel_diff, abs_diff / abs(other_value))
+                if isinstance(approx_value.expected, Number):
+                    abs_diff = abs(approx_value.expected - other_value)
```

Comment:
mypy is complaining in this part
``Unsupported operand types for - ("Number" and "float")``

should I disregards this or better to change to 
```python
if isinstance(approx_value.expected, (Decimal, Complex)):
```
?

@nicoddemus

Reply:
Let's use `(Complex, Decimal)` as we do in other parts of the code. 👍